google-oauth-client/src/test/java/com/google/api/client/auth/openidconnect/IdTokenVerifierTest.java

```diff
@@ -151,4 +184,166 @@ public void testMissingAudience() {
             .build();
     assertFalse(verifier.verify(idToken));
   }
+
+  public void testVerifyEs256TokenPublicKeyMismatch() throws Exception {
+    // Mock HTTP requests
+    HttpTransportFactory httpTransportFactory =
+        new HttpTransportFactory() {
+          @Override
+          public HttpTransport create() {
+            return new MockHttpTransport() {
+              @Override
+              public LowLevelHttpRequest buildRequest(String method, String url)
+                  throws IOException {
+                return new MockLowLevelHttpRequest() {
+                  @Override
+                  public LowLevelHttpResponse execute() throws IOException {
+                    MockLowLevelHttpResponse response = new MockLowLevelHttpResponse();
+                    response.setStatusCode(200);
+                    response.setContentType("application/json");
+                    response.setContent("");
+                    return response;
+                  }
+                };
+              }
+            };
+          }
+        };
+    IdTokenVerifier tokenVerifier =
+        new IdTokenVerifier.Builder()
+            .setClock(FIXED_CLOCK)
+            .setHttpTransportFactory(httpTransportFactory)
+            .build();
+
+    try {
+      tokenVerifier.verify(IdToken.parse(JSON_FACTORY, ES256_TOKEN));
+    } catch (VerificationException ex) {
+      assertTrue(ex.getMessage().contains("Error fetching PublicKey"));
+      return;
+    }
+    throw new Exception("Should have failed verification");
```

Comment:
My 2 cent:

I would put `fail("Should have failed verification");` immediately after tokenVerifier.verify